File: document_manager/document_manager/report/transactions_statistics/transactions_statistics.js
// Copyright (c) 2016, masonarmani and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Transactions Statistics"] = {
	"filters": [{
			"fieldname": "from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"reqd":1,
			"default":frappe.datetime.get_today()
		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"reqd":1,
			"default":frappe.datetime.add_days(frappe.datetime.get_today(),7)
		}
	]
};

File: document_manager/document_manager/report/transactions_statistics/transactions_statistics.py
# Copyright (c) 2013, masonarmani and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns = [
		{
			"fieldname": "Document",
			"label": "Document",
			"fieldtype": "Data",
			"width": 110
		},{
			"fieldname": "Transaction Count",
			"label": "Transaction Count",
			"fieldtype": "Float",
			"width": 200
		}
	]

	data = []
	conditions = ""

	if filters.get('from_date') and filters.get('to_date'):
		conditions += "posting_date between '{from_date}' and '{to_date}'"

	query = "select count(name) from `{0}` where {conditions}"

	for table in ["tabPayment Entry", "tabSales Invoice","tabPurchase Receipt","tabJournal Entry"]:
		res = frappe.db.sql(query.format(table, conditions=conditions.format(**filters)))
		data.append((table.replace("tab", ""), res[0][0], ""))

	return columns, data
